### DISCUSSION TITLE
non browser Rhino engine running xmldom

### DISCUSSION THREAD

**Discussion by lmcdasi** [xmldom/xmldom#423] [Q&A]
Hi,

   I'm trying to use xmldom in a non browser Rhino engine. I'm getting the following error: `Exception in thread "main" org.mozilla.javascript.EcmaError: TypeError: [object Object] is not a function, it is object. (<cmd>#3)`
   when executing `const parser = new DomParser();`

   Any suggestions if I can accomplisged that ? Thx.

   I have added the 'js' scripts in the src/main/resource & I used: 

```
    // ...
    Scriptable jsArgs = Context.toObject(xPathRegex, scope);
    Scriptable contentArgs = Context.toObject(content, scope);
    Scriptable xPathResult = Context.toObject(XPathResult.UNORDERED_NODE_ITERATOR_TYPE, scope);
	        
     Require require = new Require(cx, cx.initStandardObjects(),
	                new StrongCachingModuleScriptProvider(
	                        new UrlModuleSourceProvider(Collections.singleton(
	                                xpath2.getDirectory()), null)), null, null, true);
    require.install(scope);
    ScriptableObject.putProperty(scope, "xPathRegex", jsArgs);
    ScriptableObject.putProperty(scope, "content", contentArgs);
    ScriptableObject.putProperty(scope, "xpathResult", xPathResult);
	        
    String multiCode="const DomParser = require(\"dom.js\");" + "\r\n" +
        			     "function xPathEvaluate(){" + "\r\n" +
	                         "const parser = new DomParser();" + "\r\n" +
	        			     //"const document = parser.parseFromString(content);" + "\r\n" +
	                         "return \"a\";}" + "\r\n" + 
	                         "xPathEvaluate()";
	        
    final String results = (String) cx.evaluateString(scope, multiCode, "<cmd>", 1, null);
    	
    cx.exit();
```

**Comment by karfau** [comment]
The thing that you named `DomParser` is everything that is exported by dom.js.

In case this is referring to https://github.com/xmldom/xmldom/blob/5eb649e00aeaaf016cad313f12ef0da02b563a1f/lib/dom.js#L1527 it exports an object that contains multiple things but no `DomParser`.

Our main file [index.js](https://github.com/xmldom/xmldom/blob/5eb649e00aeaaf016cad313f12ef0da02b563a1f/lib/index.js#L4) (and also dom-parser.js) provides an export called `DOMParser` (different casing!).

Also for those exports you need to access them by either destrucuturing them (if your environment supports it)
```js
const {DOMParser} = require(…);
new DOMParser().parseFromString(content)
```
Or accessing it as a property:
```js
const xmldom = require(…);
new xmldom.DOMParser().parseFromString(content);
```

If dom.js is your code please provide it for further analysis if required.
